python/sglang/srt/layers/attention/intel_amx_backend.py

```diff
@@ -0,0 +1,128 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING
+
+import torch
+
+from sglang.srt.layers.attention import AttentionBackend
+from sglang.srt.model_executor.forward_batch_info import ForwardBatch
+
+if TYPE_CHECKING:
+    from sglang.srt.layers.radix_attention import RadixAttention
+    from sglang.srt.model_executor.model_runner import ModelRunner
+
+
+class IntelAMXAttnBackend(AttentionBackend):
+    def __init__(self, model_runner: ModelRunner):
+        from sgl_kernel.cpu import decode_attention, extend_attention
+
+        super().__init__()
+        self.forward_metadata = None
+        self.device = model_runner.device
+
+        self.num_head = (
+            model_runner.model_config.num_attention_heads // model_runner.tp_size
```

Comment:
what if tp_size == 3?

Reply:
is this covered by https://github.com/sgl-project/sglang/commit/61003bbdadfd6e558ff600c908493d4d10456fca?